blazor/scheduler/appointments.md

```diff
@@ -477,6 +477,63 @@ By default, the scheduler will render the overlapping events based on the start
+
+<SfSchedule TValue="AppointmentData" AllowOverlap="false" Height="550px" @bind-SelectedDate="@CurrentDate">
+    <ScheduleEventSettings DataSource="@DataSource"></ScheduleEventSettings>
+    <ScheduleEvents TValue="AppointmentData" OnPopupOpen="OnPopupOpen"></ScheduleEvents>
+    <ScheduleViews>
+        <ScheduleView Option="View.Day"></ScheduleView>
+        <ScheduleView Option="View.Week"></ScheduleView>
+        <ScheduleView Option="View.WorkWeek"></ScheduleView>
+        <ScheduleView Option="View.Month"></ScheduleView>
+        <ScheduleView Option="View.Agenda"></ScheduleView>
+    </ScheduleViews>
+</SfSchedule>
+
+@code{
+    DateTime CurrentDate = new DateTime(2024, 12, 2);
+    List<AppointmentData> DataSource = new List<AppointmentData>
+    {
+        new AppointmentData { Id = 1, Subject = "Board Meeting", StartTime = new DateTime(2024, 12, 1, 9, 30, 0),
+        EndTime = new DateTime(2024, 12, 1, 11, 0, 0) },
+        new AppointmentData { Id = 2, Subject = "Annual Conference", StartTime = new DateTime(2024, 12, 2, 10, 0, 0),
+        EndTime = new DateTime(2024, 12, 2, 11, 0, 0) },
+        new AppointmentData { Id = 3, Subject = "Tech Symposium", StartTime = new DateTime(2024, 12, 2, 10, 30, 0),
+        EndTime = new DateTime(2024, 12, 2, 11, 30, 0) },
+        new AppointmentData { Id = 4, Subject = "Client Meeting", StartTime = new DateTime(2024, 12, 3, 12, 0, 0),
+        EndTime = new DateTime(2024, 12, 3, 13, 0, 0) },
+        new AppointmentData { Id = 5, Subject = "Project Review", StartTime = new DateTime(2024, 12, 4, 13, 0, 0),
+        EndTime = new DateTime(2024, 12, 4, 15, 0, 0) },
+        new AppointmentData { Id = 6, Subject = "Strategy Session", StartTime = new DateTime(2024, 12, 6, 9, 30, 0),
+        EndTime = new DateTime(2024, 12, 6, 11, 0, 0) },
+    };
+    public void OnPopupOpen(PopupOpenEventArgs<AppointmentData> args)
+    {
+        List<AppointmentData> overlappingData = args.OverlapCollection;
+    }
+    public class AppointmentData
+    {
+        public int Id { get; set; }
+        public string Subject { get; set; }
+        public DateTime StartTime { get; set; }
+        public DateTime EndTime { get; set; }
+    }
+}
+```
+![Blazor Scheduler with restricted Overlapping Events](images/blazor-scheduler-restrict-overlapping-events.png)
+
+### Limitations
+* On initial load, the scheduler prioritizes non-overlapping events based on their longer duration and all-day status.
+* For recurring appointments, at the initial load, if there are conflicts within a series, the scheduler will display all occurrences except for the conflicting one.
+* When a user edits, saves, or drops appointments, the scheduler checks for overlaps. If a conflict is detected, it prevents the action and displays a conflict alert to the user.
+* When creating or editing a recurrence series dynamically, if a conflict is detected, the scheduler won't allow any occurrences of the conflicting series.
+
```

Comment:
Limitations
Initial Load Behavior: When the scheduler loads initially, it prioritizes non-overlapping events based on their duration and all-day status. Events with longer durations and those marked as all-day are given higher priority to avoid overlaps.

Recurring Appointments: During the initial load, if there are conflicts within a recurring appointment series, the scheduler will display all occurrences of the series, except for the conflicting instance.

Event Modifications: When a user edits, saves, or drops appointments, the scheduler checks for potential overlaps. If a conflict is found, the action is prevented, and a conflict alert is shown to the user to address the issue.

Dynamic Recurrence Series Creation or Editing: When a user creates or edits a recurrence series dynamically, the scheduler will not allow any occurrences of the series if a conflict is detected within the series itself.